pkg/migrate/azure/dns.go

```diff
@@ -0,0 +1,1335 @@
+		}
+		if rs.MxRecords != nil && len(*rs.MxRecords) > 0 {
+			showLegacyMxRecords(rs.MxRecords, zoneName, *rs.Fqdn, ttl)
+		}
+		if rs.CnameRecord != nil {
+			showLegacyCnameRecord(rs.CnameRecord, zoneName, *rs.Fqdn, ttl)
+		}
+		if rs.SrvRecords != nil && len(*rs.SrvRecords) > 0 {
+			showLegacySrvRecords(rs.SrvRecords, zoneName, *rs.Fqdn, ttl)
+		}
+		if rs.TxtRecords != nil && len(*rs.TxtRecords) > 0 {
+			showLegacyTxtRecords(rs.TxtRecords, zoneName, *rs.Fqdn, ttl)
+		}
+		if rs.ARecords != nil && len(*rs.ARecords) > 0 {
+			showLegacyARecords(rs.ARecords, zoneName, *rs.Fqdn, ttl)
+		}
+		if rs.AaaaRecords != nil && len(*rs.AaaaRecords) > 0 {
+			showLegacyAaaaRecords(rs.AaaaRecords, zoneName, *rs.Fqdn, ttl)
+		}
+		if rs.PtrRecords != nil && len(*rs.PtrRecords) > 0 {
+			showLegacyPtrRecords(rs.PtrRecords, zoneName, *rs.Fqdn, ttl)
+		}
+	}
+}
+
+// DoListLegacyZone Retrives legacy zone info and prints it out
+func DoListLegacyZone(resourceGroup string, legacyZone string, public bool) error {
+	session, err := azconfig.GetSession()
+	if err != nil {
+		return err
+	}
+
+	legacyDNSClient := newLegacyDNSClient(session, resourceGroup, public)
+
+	// Serialize/Deserialize to make deep copy
+	szone, err := legacyDNSClient.GetZoneSerialized(legacyZone)
+	if err != nil {
+		return err
+	}
+
+	legacyDNSZoneInfo, err := deserializeLegacyZone(szone)
+	if err != nil {
+		return err
+	}
+
+	doListLegacyZoneDeserialized(legacyDNSZoneInfo)
+	return nil
+}
+
+// Prints out a single private zone
+func doListPrivateZoneDeserialized(privateDNSZoneInfo *privateDNSZoneInfo) {
+	logrus.Debug(spew.Sdump(privateDNSZoneInfo))
+
+	zoneName := safeString(privateDNSZoneInfo.zone.Name)
+
+	logrus.Printf("PrivateZone: %s", zoneName)
+	for _, rs := range privateDNSZoneInfo.recordsets {
+		ttl := int64(30)
+		if rs.TTL != nil {
+			ttl = *rs.TTL
+		}
+
+		if rs.SoaRecord != nil {
+			showPrivateSoaRecord(rs.SoaRecord, zoneName, *rs.Fqdn, ttl)
+		}
+		if rs.MxRecords != nil && len(*rs.MxRecords) > 0 {
+			showPrivateMxRecords(rs.MxRecords, zoneName, *rs.Fqdn, ttl)
+		}
+		if rs.CnameRecord != nil {
+			showPrivateCnameRecord(rs.CnameRecord, zoneName, *rs.Fqdn, ttl)
+		}
+		if rs.SrvRecords != nil && len(*rs.SrvRecords) > 0 {
+			showPrivateSrvRecords(rs.SrvRecords, zoneName, *rs.Fqdn, ttl)
+		}
+		if rs.TxtRecords != nil && len(*rs.TxtRecords) > 0 {
+			showPrivateTxtRecords(rs.TxtRecords, zoneName, *rs.Fqdn, ttl)
+		}
+		if rs.ARecords != nil && len(*rs.ARecords) > 0 {
+			showPrivateARecords(rs.ARecords, zoneName, *rs.Fqdn, ttl)
+		}
+		if rs.AaaaRecords != nil && len(*rs.AaaaRecords) > 0 {
+			showPrivateAaaaRecords(rs.AaaaRecords, zoneName, *rs.Fqdn, ttl)
+		}
+		if rs.PtrRecords != nil && len(*rs.PtrRecords) > 0 {
+			showPrivatePtrRecords(rs.PtrRecords, zoneName, *rs.Fqdn, ttl)
+		}
+	}
+
+}
+
+// DoListPrivateZone Retrives private zone info and prints it out
+func DoListPrivateZone(resourceGroup string, privateZone string) error {
```

Comment:
also this `Do*` is really not necessary, `ListPrivateZone` seems like good enough.

Reply:
I've renamed everything to not have Do*.